docs/data/date-pickers/custom-field/custom-behavior/MaskedMaterialTextField.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import { useRifm } from 'rifm';
+import TextField from '@mui/material/TextField';
+import useControlled from '@mui/utils/useControlled';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import {
+  DatePicker,
+  DatePickerProps,
+  DatePickerFieldProps,
+} from '@mui/x-date-pickers/DatePicker';
+import { useSplitFieldProps, useParsedFormat } from '@mui/x-date-pickers/hooks';
+import { useValidation, validateDate } from '@mui/x-date-pickers/validation';
+
+const MASK_USER_INPUT_SYMBOL = '_';
+const ACCEPT_REGEX = /[\d]/gi;
```

Comment:
nit: not so sure about this ... `i` definitely isn't needed, but are you intending to only check for a single digit here?

Reply:
I kept the regexp that we used in v5.
This code came from the lab and I never really touched it a lot to be honested :laughing: 

https://github.com/mui/mui-x/blob/v5.x/packages/x-date-pickers/src/internals/hooks/useMaskedInput.tsx#L26

But we don't have only digits, we have letters, at least for the meridiem.

Reply:
oh right ... we could potentially improve this then and use a different regex to exclude anything else: `[\d]|(a|p)m`